File: List/merge_two_sorted_list.cpp
#include<bits/stdc++.h>
using namespace std;

struct node{
    int data;
    struct node *next;
    
    node(int x){
        this->data = x;
        this->next = NULL;
    }
};

typedef struct node *N;

N insert(N start,int x){
    N t = new node(x);

    if(!start){
        return t;
    }

    N c = start;

    while(c->next!=NULL){
        c=c->next;
    }

    c->next = t;

    return start;
}

N merge(N start1, N start2){
    N tmp, s;

    if(start1->data < start2->data){
        s = start1;
        tmp = start1;
        start1=start1->next;
    }else{
        s = start2;
        tmp = start2;
        start2=start2->next;
    }

    while(start1 and start2){
        if(start1->data < start2->data){
            tmp->next = start1;
            tmp = tmp->next;
            start1=start1->next;
        }else{
            tmp->next = start2;
            tmp = tmp->next;
            start2=start2->next;
        }
    }

    if(start1){
        tmp->next = start1;
    }else{
        tmp->next = start2;
    }

    return s;
}

void display(N start){
    N t=start;

    while(t){
        cout << t->data << endl;
        t=t->next;
    }

}

int main(){
    N start1 = NULL;
    N start2 = NULL;

    start1 = new node(10);
    start1->next = new node(20);
    start1->next->next = new node(30);

    start2 = new node(15);
    start2->next = new node(25);

    cout << "Merged Linked List :- " << endl;
    
    display(merge(start1,start2));

    return 0;
}


/* This code is contributed by Vittal AB*/

File: List/merge_two_sorted_list.py
from re import L


class node:
    def __init__(self, data):
        self.data = data
        self.next = None


class ListNode:

    def __init__(self):
        self.head = None

    def insert(self, val):
        if self.head is None:
            self.head = node(val)
        else:
            temp = self.head

            while temp.next is not None:
                temp = temp.next

            temp.next = node(val)

    def display(self, s):
        temp = s

        while temp is not None:
            print(temp.data)
            temp = temp.next
    
    def merge(self, start1, start2):
        tmp = node(0)
        s = node(0)

        if start1.data < start2.data:
            s = start1
            tmp = start1
            start1 = start1.next
        else:
            s = start2
            tmp = start2
            start2 = start2.next


        while start1 and start2:

            if start1.data < start2.data:
                tmp.next = start1
                tmp = tmp.next
                start1 = start1.next
            else:
                tmp.next = start2
                tmp = tmp.next
                start2 = start2.next
        
        if start1:
            tmp.next = start1
        else:
            tmp.next = start2

        self.display(s)


if __name__=="__main__":
    l1 = ListNode()
    l2 = ListNode()

    l1.insert(10)
    l1.insert(20)
    l1.insert(30)

    l2.insert(15)
    l2.insert(25)

    l1.merge(l1.head,l2.head)


# /* This code is contributed by Vittal AB*/